feat: add an opt-in linter job to the main workflow (#8)
fastify/workflows: README.md

Included in this repo is a [basic workflow](.github/workflows/plugins-ci.yml) for use across the majority of plugins, as well as variants with service containers.

### Enable workflow Linter job

> By setting the `lint` value to `true` when using the [basic workflow](.github/workflows/plugins-ci.yml) the CI will first run the linter job.

Reviewer: ```suggestion
By setting the `lint` option to `true` when using the [basic workflow](.github/workflows/plugins-ci.yml) the CI will first run the linter job once.

```

__Example:__ running the linter job first with the [basic workflow](.github/workflows/plugins-ci.yml)

```yml
name: CI

on:
  push:
    paths-ignore:
      - 'docs/**'
      - '*.md'
  pull_request:
    paths-ignore:
      - 'docs/**'
      - '*.md'

jobs:

>   call-reuseable-workflow:

Reviewer: easier copy-paste

```suggestion
  test:
```

    uses: fastify/workflows/.github/workflows/plugins-ci.yml@v2
    with:

>       lint: true

Reviewer: ```suggestion
      lint: false # default `true`
```

Reviewer 2: The example needs updating to remove the `lint` input usage, as it's no longer in place in this PR.

Author: added back the lint input in the last iteration.
